.github/workflows/ci.yml

```diff
@@ -14,21 +14,29 @@ jobs:
     runs-on: ${{ matrix.platform }}
 
     steps:
-    - uses: actions/checkout@v4
+      - uses: actions/checkout@v4
 
-    - name: Setup Maven cache
-      uses: actions/cache@v4
-      with:
-        path: ~/.m2/repository
-        key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
-        restore-keys: |
-          ${{ runner.os }}-maven-
+      - name: Setup Maven cache
+        uses: actions/cache@v4
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
 
-    - name: Set up JDK 11
-      uses: actions/setup-java@v4
-      with:
-        java-version: '11'
-        distribution: 'corretto'
+      - name: Set up JDK 11
+        uses: actions/setup-java@v4
+        with:
+          java-version: '11'
+          distribution: 'corretto'
 
-    - name: Build with Maven
-      run: mvn -e --no-transfer-progress clean install
\ No newline at end of file
+      - name: Build with Maven
+        run: mvn -e --no-transfer-progress clean install
+
+      - name: Check for changes
+        shell: bash
+        run: |
+          git diff --exit-code || (
+            echo "Code changes detected - run 'mvn clean verify' locally"
```

Comment:
i guess it is a good place to use GitHub Actions error annotation (::error::), so it can be highlighted in UI

```
echo "::error::Code changes detected - run 'mvn clean verify' locally"
```

Code changes detected - run 'mvn clean verify' locally

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#setting-an-error-message

btw I've never tried it before :)